back_end/index.js

```diff
@@ -419,6 +419,163 @@ app.get('/fornecedor/:id', (req, res) => {
   console.log('get /fornecedor/' + fornecedorId + ' executado. Fornecedor retornado com sucesso!');
 });
   
+
+//alteração Erik e Marcos//
+
+app.get('/usuario', (req, res) => {
+  db.query('SELECT * FROM usuario', (err, results) => {
+    if (err) {
+      res.status(500).send('Erro retornando usuario: ' + err);
+      return;
+    }
+    res.json(results);
+  });
+  console.log('get /usuario executado. Usuarios retornados com sucesso!');
+});
+   
+/* POST */
+app.post('/usuario/create', (req, res) => {
+    const { login, nome, privilegios, senha } = req.body;
+    const query = `INSERT INTO usuario (login, nome, privilegios, senha) VALUES (?, ?, ?, ?)`;
+    const values = [login, nome, privilegios, senha];
+    db.query(query, values, (err, result) => {
+    if (err) {
+      res.status(500).send('Erro criando usuario: ' + err);
+      return;
+    }
+    const postId = result.insertId;
+    db.query('SELECT * FROM usuario WHERE id_usuario = ?', postId, (err, result) => {
+      if (err) {
+        res.status(500).send('Erro recuperando usuario criado: ' + err);
+        return;
+      }
+      res.status(201).json(result[0]);
+    });
+  });
+  console.log('post executado. Usuario criado com sucesso!');
+});
+  
+/* GET */
+app.get('/usuario/:id', (req, res) => {
+  const usuarioId = req.params.id;
+  db.query('SELECT * FROM usuario WHERE id_usuario = ?', usuarioId, (err, result) => {
+    if (err) {
+      res.status(500).send('Erro recuperando usuario: ' + err);
+      return;
+    }
+    if (result.length === 0) {
+      res.status(404).send('Usuario não encontrado: ' + err);
+      return;
+    }
+    res.json(result[0]);
+  });
+  console.log('get /usuario/' + usuarioId + ' executado. Usuario retornado com sucesso!');
+});
+
+/* PUT */
+app.put('/usuario/:id', (req, res) => {
+  const usuarioId = req.params.id;
+  const { login, nome, privilegios, senha } = req.body;
+  const query = `UPDATE usuario SET login = ?, nome = ?, privilegios = ?, senha =?`;
+  const values = [login, nome, privilegios, senha, usuarioId];
+    db.query(query, values, err => {
```

Comment:
alterando a senha do usuario? n tem um endpoint pra isso? se for fazer mesmo teria q criptografar dnv assim como na criação